packages/contracts/test/utils/combinatorial_sets.ts

```diff
@@ -0,0 +1,25 @@
+import { BigNumber } from '@0xproject/utils';
+import * as _ from 'lodash';
+
+export const BigNumbers = [];
+
+const MAX_256_BIT_NUMBER = new BigNumber(2).pow(256).minus(1);
+
+export const positiveNaturalBigNumbers = [
+    new BigNumber(0),
+    new BigNumber(1),
+    new BigNumber(2),
+    new BigNumber(2).pow(64),
+    MAX_256_BIT_NUMBER,
+    MAX_256_BIT_NUMBER.div(2).floor(),
+    MAX_256_BIT_NUMBER.sqrt().floor(),
+];
+
+export const negativeNaturalNumbers = _.map(positiveNaturalBigNumbers, n => n.neg());
```

Comment:
Negative numbers in Solidity use twos-complement, which means their values range from `-2**128 ... 2**128-1`. Note that in twos complement there is always one more negative number than positive (not counting zero).

Values like `MAX_256_BIT_NUMBER` and `-MAX_256_BIT_NUMBER` do not exist in singed integer.

It's best not to think of them as "natural numbers". Such a thing does not exist in Solidity. In Solidity there are several kinds of numbers: `uint256`, `uint248` .. `uint8`, `sint256`, ... `sint8`. All of them behave according to their own set of rules and should have their own specific tests sets. Similarly other data types such as `bool`, `address`,`byte1` ... `byte32`, `bytes`, `string` should also have their own test sets. Of course we only need to implement the ones that we actually use.

Reply:
👍I'm going to change our approach here to use a set of inputs for each type.